### ISSUE TITLE
Unhelpful "std::marker::Sized` is not satisfied" error

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/rust#33869]
Compiling this program:

``` rust
fn main() {        
    let b = bar;   
    foo(&b);       
}                  

fn foo(_: &[i32]) {
}
```

yields

```
foo.rs:2:13: 2:16 error: unresolved name `bar` [E0425]
foo.rs:2     let b = bar;
                     ^~~
foo.rs:2:13: 2:16 help: run `rustc --explain E0425` to see a detailed explanation
foo.rs:2:9: 2:10 error: the trait bound `[i32]: std::marker::Sized` is not satisfied [E0277]
foo.rs:2     let b = bar;
                 ^
foo.rs:2:9: 2:10 help: run `rustc --explain E0277` to see a detailed explanation
foo.rs:2:9: 2:10 note: `[i32]` does not have a constant size known at compile-time
foo.rs:2:9: 2:10 note: all local variables must have a statically known size
error: aborting due to previous error
```

The first error, the unresolved name, is legitimate, but the next is just extra information that's not too helpful and is likely fixed once the previous error is fixed.

**Comment by alexcrichton** [comment]
cc @nrc

**Comment by nrc** [comment]
Interesting, we're probably checking that `ty::Error: Sized`, which is nonsense and shouldn't cause an error.

**Comment by eddyb** [comment]
@nrc On the contrary, `ty::Error` is the solution, and it's not getting used; this is a duplicate of #33575.

**Comment by Mark-Simulacrum** [comment]
Closing in favor of #33575.
